plugin/src/main/kotlin/com/ritense/plugin/domain/PluginConfiguration.kt

```diff
@@ -39,9 +43,9 @@ class PluginConfiguration(
     var title: String,
     @Type(type = "com.vladmihalcea.hibernate.type.json.JsonType")
     @Column(name = "properties", columnDefinition = "JSON")
-    var properties: JsonNode? = null,
+    val properties: JsonNode? = null,
     @JoinColumn(name = "plugin_definition_key", updatable = false, nullable = false)
-    @ManyToOne(fetch = FetchType.LAZY)
+    @ManyToOne(fetch = FetchType.EAGER)
```

Comment:
Why was this necessary? Do we always need the definition?

Reply:
Every time we load a plugin configuration we need the definition to identify any secrets we need to decrypt. Since we access this variable every time there's no reason to keep this lazy.